imports/api/resources/resource.methods.js

```diff
@@ -3,106 +3,174 @@ import { getHost } from '../@/shared';
 import { isContributorOrAdmin } from '../@users/user.roles';
 import Hosts from '../@hosts/host';
 import Resources from './resource';
+import Activities from '../activities/activity';
 
+// RESOURCE METHOD VALIDATIONS
+function validateUser(user, currentHost) {
+  if (!user || !isContributorOrAdmin(user, currentHost)) {
+    throw new Meteor.Error('You are not allowed');
+  }
+  return true;
+}
+function validateLabel(label, host, resourceId) {
+  // set resource query
+  let resourceQuery = { host, label };
+  if (resourceId) resourceQuery._id = { $ne: resourceId };
+  // validate label
+  if (label.length < 3) {
+    throw new Meteor.Error('Resource name is too short. Minimum 3 letters required');
+  } else if (Resources.find(resourceQuery).fetch().length > 0) {
+    throw new Meteor.Error('There already is a resource with this name');
+  }
+  return true;
+}
+// FETCH COMBO RESOURCES
+function fetchComboResources(resource) {
+  const resourcesForCombo =  Resources.find(
+    { _id : { $in : resource.resourcesForCombo } }, 
+    { fields: { label: 1 } }
+  ).fetch();
+  resource.resourcesForCombo = resourcesForCombo;
+  return resource;
+}
+// RESOURCE METHODS
 Meteor.methods({
   getResources() {
     const host = getHost(this);
+    const sort = { createdAt: -1 };
+    const fields = Resources.publicFields;
+    const resources = Resources.find({ host }, { sort, fields }).fetch();
+    resources.forEach(resource => {
+      if (resource.isCombo) 
+        resource = fetchComboResources(resource);
+    });
+    resources.forEach(resource => {
+      if (resource.userId) {
+        resource.user = {} = Meteor.users.findOne(resource.userId, { fields: { username: 1, avatar: { src: 1 } }});
+        resource.user.avatar = resource.user.avatar.src;
+        delete resource.userId;
+      }
+    });
+    return resources;
+  },
 
-    return Resources.find({ host }, { sort: { creationDate: 1 } }).fetch();
+  getResourceLabels() {
```

Comment:
Yeah getResources does nearly the same. I did a seperate method cause form component doesnt need anything more than id or label of resources so I made a custom method for selectbox data only instead of calling all resources data method. But ofcource we dont have to use that way.